streamwebs_frontend/streamwebs/tests/models/test_water_quality.py

```diff
@@ -0,0 +1,136 @@
+            'salinity': models.DecimalField,
+            'conductivity': models.DecimalField,
+            'total_solids': models.DecimalField,
+            'bod': models.DecimalField,
+            'ammonia': models.DecimalField,
+            'nitrite': models.DecimalField,
+            'nitrate': models.DecimalField,
+            'phosphates': models.DecimalField,
+            'fecal_coliform': models.DecimalField,
+            'notes': models.TextField,
+            'id': models.AutoField
+        }
+
+        # Where does 'Vernier' and 'Manual' fit into the model?
+        # Celcius and Fahrenheit?
+        # Sample number??
+
+        self.optional_fields = {
+            'conductivity',
+            'total_solids',
+            'bod',
+            'ammonia',
+            'nitrite',
+            'nitrate',
+            'phosphates',
+            'fecal_coliform',
+            'notes',
+        }
+
+    def test_fields_exist(self):
+        model = apps.get_model('streamwebs', 'water_quality')
+        for field, field_type in self.expected_fields.items():
+            self.assertEqual(
+                field_type, type(model._meta.get_field(field)))
+
+    def test_no_extra_fields(self):
+        fields = list(set(chain.from_iterable(
+            (field.name, field.attname) if hasattr(field, 'attname') else
+            (field.name,) for field in Water_Quality._meta.get_fields()
+            if not (field.many_to_one and field.related_model is None)
+        )))
+        self.assertEqual(sorted(fields), sorted(self.expected_fields.keys()))
+
+    def test_optional_fields(self):
+        apps.get_model('streamwebs', 'site')
+        for field in self.optional_fields:
+            self.assertEqual(
+                Water_Quality._meta.get_field(field).blank, True)
+
+    def test_datasheet_ManyToOne(self):
+        '''Tests that the new datasheet correctly corresponds to the
+           specified site'''
+        site = Site.objects.create_site('test', 'some_type', 'some_slug')
+        waterq = Water_Quality.objects.create(site=site, 
+                                              DEQ_wq_level='A',
+                                              date='2016-06-01',
+                                              school='a', teacher='teacher',
+                                              latitude=90, longitude=123,
+                                              fish_present='False',
+                                              live_fish=0, dead_fish=0,
+                                              water_temp=45, air_temp=65,
+                                              dissolved_oxygen=15, pH=6.7,
+                                              turbidity=0, salinity=12)
+        self.assertEqual(waterq.site.site_name, 'test')
+        self.assertEqual(waterq.site.site_type, 'some_type')
+        self.assertEqual(waterq.site.site_slug, 'some_slug')
+
+#    def test_datasheet_nonexistent_site(self):
```

Comment:
@Kennric I wrote tests to assert that datasheet creation fails when the user tries to link/associate it with a nonexistent or bad site. But looking at it again, I don't think there are any cases in which this would ever happen? Especially since the user will only be able to select from a list of existing sites.

Am I correct on that front? Or should I try to implement these tests?